### DISCUSSION TITLE
Go To Definition behavior

### DISCUSSION THREAD

**Discussion by koby08** [REditorSupport/vscode-R#1298] [Q&A]
Hi,

Just installed vscode-R and I am seeing a (weird?) behavior when using "Go to Definition" - instead of jumping to the function definition in its source file, it jumps to a temporary file with the header being "# Generated from function body. Editing this file has no effect."

Is there a way to solve this? I am feeling I might be missing some configuration?

Thanks!

**Comment by renkun-ken** [comment]
It is because the language server thinks the function is from an R package rather than from another file in the workspace.

**Comment by koby08** [comment]
Yeah that is what I've assumed, but any idea how to overcome this? Can I somehow configure its search path or something?

**Comment by koby08** [comment]
Hi, had some progress with this and realized that if I Open Folder the directory with the R files directly, then everything works as expected (i.e. Go to Definition opens the source file).

The thing is that this R directory is a subdirectory in a bigger repo. Is there a way to configure the vscode-R plugin workspace dir, or tell it to look recursively?
